serial/api_client.py

```diff
@@ -0,0 +1,78 @@
+import requests
+import serial
+
+class APIClient:
+    def __init__(self, api_key, base_url):
+        """
+        Initializes a new API Client
+
+        Returns:
+        - A new API client
+
+        Once initialized, it should be passed an api_key to handle requests,
+        as well optionally receiving a station id and new base url (which defaults
+        to the default serial api url of https://api.serial.io)
+        """
+        self._api_key = api_key 
+        self._base_url = base_url 
+
+    def make_api_request(self, endpoint, method, params=None, data=None, files=None):
+        if method == "GET":
+            response = self._get(endpoint, params)
+        elif method == "POST" and not files:
+            response = self._post(endpoint, data=data)
+        elif method == "POST" and files:
```

Comment:
I'm not sure if this distinction is necessary, but I just wanted something to work.

Reply:
I think this is reasonable